iwyu_tool.py

```diff
@@ -107,12 +107,33 @@ def find_include_what_you_use():
         if os.path.isfile(full):
             return full
 
-    return None
-
 
 IWYU_EXECUTABLE = find_include_what_you_use()
 
 
+class InvocationResult(object):
```

Comment:
I experimented with a variation on this design -- I don't think `InvocationResult` conveys that this is a process in flight. Something like the below works much better for my brain:

```
class Process(object):
    def __init__(self, proc, outfile):
        self.proc = proc
        self.outfile = outfile
        self.output = None

    def poll(self):
        return self.proc.poll()

    def get_output(self):
        if not self.output:
            self.proc.wait()
            self.outfile.seek(0)
            self.output = self.outfile.read().decode("utf-8")
            self.outfile.close()

        return self.output

    @classmethod
    def start(cls, invocation, verbose):
        if verbose:
            print('# %s' % invocation)

        outfile = tempfile.TemporaryFile(prefix='iwyu')
        process = subprocess.Popen(
            invocation.command,
            cwd=invocation.cwd,
            stdout=outfile,
            stderr=subprocess.STDOUT)
        return cls(process, outfile)
```

This:
- removes the mutual dependency between this class and `Invocation`
- makes `poll` return the exit code if available -- at some point we may want to use that
- Renames get -> get_output and have it cache the resulting content and close the file

Rather than jumping hoops through `Invocation.run_iwyu`, I just had the processing loop call `Process.start(invocation, verbose)` for every invocation eligible for start.